driven/storage/adapter.go

```diff
@@ -1203,19 +1203,26 @@ func (sa *Adapter) FindAccountsByParams(searchParams map[string]interface{}, app
 	for i, appOrg := range appOrgs {
 		appOrgIDs[i] = appOrg.ID
 	}
-	searchParams["app_org_id"] = appOrgIDs
-	filter := sa.getFilterForParams(searchParams)
 
-	var accounts []map[string]interface{}
 	options := options.Find()
-	options.SetLimit(int64(limit))
-	options.SetSkip(int64(offset))
+	var filter bson.M
+	if rangeIndex == "" {
+		searchParams["app_org_id"] = appOrgIDs
+		filter = sa.getFilterForParams(searchParams)
```

Comment:
We need to apply the search params to the filter in both cases here. With the current setup it will ignore the query filters if you attempt to do pagination using the new method.